### ISSUE TITLE
False positive no-member for argparse on Python 3.9

### ISSUE THREAD

**Issue by shadchin** [PyCQA/pylint#3942]
### Steps to reproduce
Here's a minimal repro:
```python
"""XXX"""
import argparse


def main():
    """XXX"""
    parser = argparse.ArgumentParser()
    parser.add_argument('cmd_args', type=str, nargs=argparse.REMAINDER, help='arguments')

    args = parser.parse_args([])
    print(args.cmd_args)
```

### Current behavior
Default config
```
--------------------------------------------------------------------
Your code has been rated at 10.00/10 (previous run: 0.00/10, +10.00)
```

With `--ignore-on-opaque-inference=n`

```
************* Module test
test.py:11:10: E1101: Instance of 'Namespace' has no 'cmd_args' member (no-member)

-------------------------------------------------------------------
Your code has been rated at 1.67/10 (previous run: 10.00/10, -8.33)
```

### Expected behavior
As the `cmd_args` attribute is present, I think `pylint` should not warn that it's not present.

I no have warn on Python 3.8 with/without `--ignore-on-opaque-inference=n`.

### pylint --version output
```
pylint 2.6.0
astroid 2.4.2
Python 3.9.0 (default, Oct  6 2020, 02:54:50)
[GCC 5.4.0 20160609]

```

**Comment by hippo91** [comment]
@shadchin thanks for the report. It seems to be indeed a bug that we need to fix before moving to `python3.9`

**Comment by globau** [comment]
This issue appears to impact multiple core modules, not just argparse (eg. math, socket, termios).
Is it odd that CI, which appears to be configured to run on 3.9, hasn't flagged this as an issue?

**Comment by Pierre-Sassoulas** [comment]
The tests are very generic, we're not making any lib specific tests (even for numpy). We need a test for ``--ignore-on-opaque-inference=n`` though.

**Comment by DanielNoord** [comment]
This no longer reproduces on `main`.

If you're still getting issues here consider using the `generated-members` option and setting `argparse.Namespace` there.
